packages/netlify-cms-core/src/reducers/auth.ts

```diff
@@ -0,0 +1,42 @@
+import { fromJS } from 'immutable';
+import { User } from 'netlify-cms-lib-util';
+import {
+  AUTH_REQUEST,
+  AUTH_SUCCESS,
+  AUTH_FAILURE,
+  AUTH_REQUEST_DONE,
+  LOGOUT,
+  AuthAction,
+} from '../actions/auth';
+import { StaticallyTypedRecord } from '../types/immutable';
+
+export type Auth = StaticallyTypedRecord<{
```

Comment:
I think it's better to define state slice type right in reducer file and then import it to types/redux.ts. Currently all state slices types are defined in types/redux.ts, which makes redux.ts file very bloated.